week-07-rails-continued/day-02/module-01/readme.md

```diff
@@ -1 +1,62 @@
 # <img src="https://cloud.githubusercontent.com/assets/7833470/10899314/63829980-8188-11e5-8cdd-4ded5bcb6e36.png" height="60"> Rails Auth
+
+
+| Objectives |
+| :--- |
+| Implement a User model that securely stores passwords. |
+| Build routes, controllers, and views necessary for a typical user flow. |
+| Add a `current_user` application-level helper method to keep track of the currently signed in user. |
+
+## Devise
+
+<img src="https://cloud.githubusercontent.com/assets/1329385/11758689/15df1a8c-a023-11e5-9e59-065e5bb5dd23.gif">
+
+I get it, some of you have heard of <a href="https://github.com/plataformatec/devise" target="_blank">Devise</a>. That's cool and all but they recommend that if you're new to rails:
+
+> If you are building your first Rails application, we recommend you do not use Devise. Devise requires a good understanding of the Rails Framework. In such cases, we advise you to start a simple authentication system from scratch. ... <a href="https://github.com/plataformatec/devise#starting-with-rails" target="_blank">[docs]</a>
+
+## Auth in Rails
+
+An auth flow hasn't changed from what <a href="https://github.com/sf-wdi-24/modules/blob/master/week-04-associated-data-and-users/day-02/module-01/readme.md#implementing-authentication" target="_blank">we learned in week 4</a> but with Rails we can build these route scaffolds with less work.
+
+In order to "roll our own" authentication we're going to follow <a href="https://gist.github.com/eerwitt/b36db29a025366037925" target="_blank">this tutorial</a>.
+
+Don't copy paste the tutorial, this is the most important part and to avoid copy-pasta we're going to change some core pieces.
+
+## Requirements
+
+1. Your rails app can't be named "gif_vault".
+1. Your rails app must not include any routes related to "gifs".
+1. You must be able to explain these terms:
+  * <details>
+      <summary>`bcrypt`</summary>
+
+      > The bcrypt function is the default password hash algorithm for BSD and other systems ... <a href="https://en.wikipedia.org/wiki/Bcrypt" target="_blank">[wiki]</a>
+
+      A method of doing one-way hashes of passwords.
+    </details>
+  * <details>
+      <summary>CSRF Token</summary>
+
+      > Synchronizer token pattern is a technique where a token, secret and unique value for each request, is embedded by the web application in all HTML forms and verified on the server side. The token may be generated by any method that ensures unpredictability and uniqueness ... <a href="https://en.wikipedia.org/wiki/Cross-site_request_forgery#Prevention" target="_blank">[wiki]</a>
+
+    </details>
+  * `has_secure_password`
+
+## Challenges
+
+1. Don't let a logged in user signup or login again.
+1. Implement `password_confirmation` for new users.
+1. Create the ability to "remember me" for users logging in.
+  * How long do you remember them for?
+1. Add ability to reset a password.
+1. Add a login form to a header navigation.
```

Comment:
I had no idea what this was saying at first. Maybe say, "Instead of having a separate login view, give your site a navbar, and add the login form to the nav".